### PR TITLE
SCRUM-3156 Update curation system to align with LinkML v1.8.0

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Audited
+@Entity
+@Data
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Inheritance(strategy = InheritanceType.JOINED)
+@Schema(name = "association", description = "Annotation class representing a disease annotation")
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java]: Need to change this name

File: src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	private String modEntityId;
+	
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "modInternalId_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@Column(unique = true)
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	private String modInternalId;
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java:83]: If these are going to be looked up by curie or any of these ids you might want indexes for them also

File: src/main/java/org/alliancegenome/curation_api/model/entities/DiseaseAnnotation.java [hunk 3]
```diff
@@ -119,26 +90,13 @@ public abstract class DiseaseAnnotation extends Association {
 	@JoinTable(indexes = { @Index(columnList = "diseaseannotation_id"), @Index(columnList = "evidencecodes_curie")})
 	private List<ECOTerm> evidenceCodes;
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/DiseaseAnnotation.java:91]: Would evidences move up also i am assuming this is precursor to pheno annotations?

File: src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+			return null;
+		}
+		
+		if (validatedDataProvider.getId() == null)
+			validatedDataProvider = dataProviderDAO.persist(validatedDataProvider);
+		
+		return validatedDataProvider;
+	}
+
+	public List<Note> validateRelatedNotes(Annotation uiEntity, Annotation dbEntity) {
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java:90]: Is this note validation code somewhere else also?

File: src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql [hunk 5]
```diff
@@ -0,0 +1,136 @@
+ -- Rename DiseaseAnnotation_updatedby_index and DiseaseAnnotation_createdby_index
+ -- Rename Association tables as Annotation and add columns
+ -- Recreate Association tables
+
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: Will need to take a closer look at this when i get to a computer

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+UPDATE annotation SET modinternalid = diseaseannotation.modinternalid FROM diseaseannotation WHERE annotation.id = diseaseannotation.id;
+UPDATE annotation SET uniqueid = diseaseannotation.uniqueid FROM diseaseannotation WHERE annotation.id = diseaseannotation.id;
+UPDATE annotation SET dataprovider_id = diseaseannotation.dataprovider_id FROM diseaseannotation WHERE annotation.id = diseaseannotation.id;
+UPDATE annotation SET singlereference_curie = diseaseannotation.singlereference_curie FROM diseaseannotation WHERE annotation.id = diseaseannotation.id;
+DELETE FROM annotation WHERE dataprovider_id IS NULL;
+
+ALTER TABLE annotation ADD CONSTRAINT annotation_singlereference_curie_fk FOREIGN KEY (singlereference_curie) REFERENCES reference (curie);
+ALTER TABLE annotation ADD CONSTRAINT annotation_dataprovider_id_fk FOREIGN KEY (dataprovider_id) REFERENCES dataprovider (id);
+ALTER TABLE annotation ADD CONSTRAINT annotation_modentityid_key UNIQUE (modentityid);
+ALTER TABLE annotation ADD CONSTRAINT annotation_modinternalid_key UNIQUE (modinternalid);
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: typically this would be named: annotation_modinternalid_uk

Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: Yeah, that's what I thought too - I was going with the format of some existing UK constraints in the database, but I prefer the _uk suffix so will change to that

File: src/main/java/org/alliancegenome/curation_api/model/entities/DiseaseAnnotation.java (hunk 3 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/DiseaseAnnotation.java:91]: The LinkML model doesn't (and didn't previously) have an evidence_codes slot for phenotype annotations.

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/model/entities/DiseaseAnnotation.java:91]: Ahh ok

File: src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java (hunk 4 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java:90]: The validation of the note object itself is handled by NoteValidator.java.  This method does some extra validation specific to annotations and some cleanup.  There is a similar, but not identical, method in AlleleValidator.java.

File: src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql (hunk 6 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: Changed now

File: src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java (hunk 2 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java:83]: Done

File: src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java]: Done

File: src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+ALTER TABLE diseaseannotation DROP CONSTRAINT uk_2ea912q3hgfs30y1wo2c868wx;
+ALTER TABLE diseaseannotation DROP CONSTRAINT uk_hvn7cmiviowf91epvfp1r67vy;
+ALTER TABLE diseaseannotation DROP CONSTRAINT uk_hlsp8ic6sxwpd99k8gc90eq4a;
+ALTER TABLE diseaseannotation DROP CONSTRAINT diseaseannotation_dataprovider_id_fk;
+ALTER TABLE diseaseannotation DROP CONSTRAINT diseaseannotation_singlereference_curie_fk;
+
+ALTER TABLE diseaseannotation_note RENAME TO annotation_note;
+ALTER TABLE annotation_note RENAME COLUMN diseaseannotation_id TO annotation_id;
+ALTER TABLE annotation_note RENAME CONSTRAINT fk52oyqivgshum07elq4opyfr5o TO annotation_note_relatednotes_id_fk;
+ALTER TABLE annotation_note RENAME CONSTRAINT uk_f3o1apeoj9un48jw6qmiihpjc TO annotation_note_relatednotes_id_key;
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: Looks like another UK here also

File: src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java (hunk 4 above)
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/services/validation/AnnotationValidator.java:90]: Ahh ok

**Review by oblodgett — APPROVED** [review]

File: src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql (hunk 7 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.25.0.1__disease_annotation_refactor.sql]: Already got that one!

**Pull request merged**